src/tsd/HttpQuery.java

```diff
@@ -114,10 +92,6 @@
   /** Deferred result of this query, to allow asynchronous processing.  */
   private final Deferred<Object> deferred = new Deferred<Object>();
 
-  /** The response object we'll fill with data */
-  private final DefaultHttpResponse response =
-    new DefaultHttpResponse(HttpVersion.HTTP_1_1, HttpResponseStatus.OK);
-
   /** The {@code TSDB} instance we belong to */
   private final TSDB tsdb;
```

Comment:
This could move to the abstract class since almost every RPC implementation would need the TSD.